tools/sof-tplgreader.py

```diff
@@ -361,14 +353,19 @@ def parse_and_set_block_keyword(block_str, tplgreader):
 
     # If run general test case on WOV pipeline or ECHO REFERENCE capture pipeline, there will be error
     # due to no feed data, and they should be blocked in general test case.
-    default_block_keyword = 'kpbm:any;type:capture & echo;'
+    default_block_keyword = 'kpbm:any;kpb:any;type:capture & echo;'
     # if no filter or block item is specified, the "block_none" here will help to
     # block nothing
     block_str = "block_none"
     # user specified block items from command line
     cmd_block_str = ret_args['block'].strip() if ret_args['block'] is not None else ''
 
-    if ret_args['filter'] is not None and 'kpbm' not in ret_args['filter'] and 'echo' not in ret_args['filter']:
+    if (
+        ret_args["filter"] is not None
+        and "echo" not in ret_args["filter"]
+        and "kpbm" not in ret_args["filter"]
+        and "kpb" not in ret_args["filter"]
```

Comment:
```suggestion
                        and not set.intersection({"echo", "kpbm", "kpb"}, ret_args["filter"])
```
Even better: give the `{"echo", "kpbm", "kpb"}` `set` a name, something like `[in]compatible_features`

Reply:
ret_args["filter"] is a string from command line, the value for -f option below.
`sof-tplgreader.py /usr/lib/firmware/intel/sof-tplg/sof-hda-generic-4ch.tplg -f 'type:playback & id:any' -b ' pcm:HDA Digital' -s 0 -e`, so set is not going to work here. How about a re match with pattern 'echo|kpbm|kpb'.